crmsh/ssh_key.py

```diff
@@ -265,3 +267,37 @@ def ensure_key_pair_exists_for_user(
             else:
                 keys.append(InMemoryPublicKey(line))
         return generated, keys
+
+
+def fetch_public_key_list(
+        host: typing.Optional[str],
+        user: str,
+        generate_key_pair: bool = False,
+        with_content: bool = False
+) -> typing.List[str]:
+    """
+    Fetch the public key list for the specified user on the specified host.
+
+    :param host: the host where the user is located. If None, the local host is assumed.
+    :param user: the user name
+    :param generate_key_pair: whether to generate a new key pair if no key pair is found,
+     default is False
+    :param with_content: whether to return the content of the public key files,
+     default is False
+
+    :return: a list of public key files if with_content is False, otherwise a list of public key strings
```

Comment:
1. It is a bad idea to return vaules in a same type with different meanings.  Please use `Key` in function signature and use `KeyFile` and `InMemoryKey` correspondingly. Or separate these 2 different usage into 2 functions. 
2. `typing.List[str]` is deprecated. Use `list[str]` instead.